sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionProcessor.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.messaging.eventhubs.models.PartitionContext;
+import reactor.core.publisher.Mono;
+
+/**
+ * The interface defining all the operations that must be supported by a single partition processor.
+ * <p>
+ * An instance of partition processor will process events only from a single partition. New instances of partition
+ * processors will be created through {@link PartitionProcessorFactory#createPartitionProcessor(PartitionContext,
+ * CheckpointManager) PartitionProcessorFactory}.
+ * </p>
+ * <p>
+ * Implementations of this interface also have the responsibility of updating checkpoints when appropriate.
+ * </p>
+ */
+public interface PartitionProcessor {
+
+    /**
+     * This method is called when this {@link EventProcessorAsyncClient} takes ownership of a new partition and before any
+     * events from this partition are received.
+     *
+     * @return a representation of the deferred computation of this call.
+     */
+    Mono<Void> initialize();
+
+    /**
+     * This method is called when a new event is received for this partition. Processing of this event can happen
+     * asynchronously.
+     *
+     * <p>
+     * This is also a good place to update checkpoints as appropriate.
+     *
```

Comment:
missing that closing `</p>` tag. Have you seen the rendered javadocs? The places where IntellIJ adds in the `<p>` may look weird as paragraphs because these are one sentence paragraphs. 

I disabled that in my IDE. Docs like the ones Jesse writes I can see using `<p>`, but this seems odd as a paragraph.